### ISSUE TITLE
[Question]: OCF > OCX - Accurate lifecycle capture for equity compensation types

### ISSUE THREAD

**Issue by jarrettq** [Open-Cap-Table-Coalition/Open-Cap-Format-OCF#513]
### Question:

Do we have all of the necessary transaction types to accurately capture/derive the state of equity pool shares tied to various types of equity compensation (e.g. RSAs, RSUs, & options)? Breaking this down into component questions further below. 

### Anything else we need to know?

NOTE: I'm attending the next LFWG session to raise this topic - posting it here as an issue so that I can provide updates in a more public forum.

In the review sessions for OCX v0.7, we covered a discussion on RSA handling that noted:

> When RSA etc is granted, should come out of Plan pool immediately on date of grant, and then become outstanding shares only after agt is signed and payment is delivered.
> When RSA has been approved but not yet executed, currently those shares would not show up in the common tab.

This was in relation to the consensus to track RSAs across both the Stock Plan and relevant Stock Ledger sheets in OCX exports to track it's progression to becoming outstanding shares on the cap table. e.g. At the point of being either approved or granted, the RSA should appear in the Stock Plan ledger. At the point of being accepted (& payment being received), the RSA also appears in the Common Stock ledger. 

Currently, `Issuance` transactions are the starting point for all security types. **One important question related to this pattern in OCF is whether the Issuance/Grant date is always the first relevant event in an equity compensation security's "lifecycle" - or whether `Board Approval` in and of itself can be the point at which shares in the equity pool are withdrawn/reserved.** Alternatively, understanding whether we should treat Board Approval & "Issuance" (Grant) dates for equity compensation as one-and-the-same is another approach. 

Below is a chart that looks outline/clarify events and their effect on a few forms of equity compensation. **One additional question here is whether the concept of `Promised Shares` is functionally equivalent to `Outstanding Options`** - which will impact the OCX Summary sheet generation (and, eventually, calculation work tied to Company Capitalization definitions). 

|  | On Board Approval | On Grant | On Acceptance | On Vesting | On Termination | On Expiration |
| --- | --- | --- | --- | --- | --- | --- |
| Options (assuming no early exercise) | Equity pool shares become `Outstanding Options` | No change - `Outstanding Options` | No change - `Outstanding Options` | No change - `Outstanding Options` | On termination, unvested options are `Forfeited` & returned to Equity Pool*. Last date to exercise established as superseding expiration date. | Beyond last date to exercise, vested & unexercised options `Expire` & return to Equity Pool. |
| RSA | Equity pool shares become `Promised Options/Shares` | ??? | Promised shares become `Outstanding Shares` | No change - `Outstanding Shares` | On termination, if granted in exchange for services, unvested shares are `Forfeited` & returned to Equity Pool. If purchased by the recipient, they are Subject to Repurchase but remain Outstanding Shares. `Repurchased` shares return to Equity Pool | N/A |
| RSU | Equity pool shares become `Promised Options/Shares` | ??? | Promised shares become ??? | Each vesting tranche, ???. Technically they become `Outstanding Shares` but the company needs to account for the grant financially.  | Unvested shares are `Forfeited` and returned to Equity Pool. | N/A |

While OCF isn't baking logic into the data models, getting alignment on these will help ensure we have the necessary data points in the format to inform OCF -> OCX translation.

**Comment by JSv4** [comment]
Thanks for the nice write-up @jarrettq. Whatever the outcome here, would also be good to capture conventions in a structured form in the XState repo. At a minimum, that could be turned into a more formal spec of validation rules, and, ideally, we'd just incorporate the rules into the state machine.

**Comment by JSv4** [comment]
My understanding here (further underscoring the need to document, document, document):

- `whether Board Approval in and of itself can be the point at which shares in the equity pool are withdrawn/reserved`
  - We don't have an event for this. Approval dates (SH and Board) are properties on the issuance. Functionally, this means that the issuance can't be modeled until approval has happened. 
- `whether the concept of Promised Shares is functionally equivalent to Outstanding Options`
  - It's been a long time, but I recall we decided not to track promised options **pre-approval** and similar pre-issuance obligations primarily because some vendors didn't track it. To the extent we're waiting on acceptance, however, and want to call those promised options, then yes they're equivalent to outstanding options.
  - IF we have a real-world state that can't be properly represented as a result - the RSUs are an interesting example
-

**Comment by JSv4** [comment]
Some Notes:

- Events
  - Does there need to be a formal distinction between board approval event and grant event. 
  - Possibly board approval event is the first event for certain securities. 
- RSAs
  - How should we treat these - as options or shares? 
  - Seems like folks are generally doing these as shares, but there doesn't seem to be a single approach.

**Comment by JSv4** [comment]
How'd the conversation go @jarrettq ? Can you share any notes or action-items?

**Comment by jarrettq** [comment]
The convo went well @JSv4 - bullet points below, I'm looking to bolster it with some more information/tables that could potentially be folded into the documentation:
- For all equity compensation security types, the earlier of either `Board Approval Date` or `Grant/Issue Date` is when shares should be deducted from the option pool and tracked as `Outstanding Options`  for the purposes of OCX. 
  - For RSAs & RSUs, after approval/grant and prior to acceptance (& payment), the grant should appear in the relevant Stock Plan ledger. 
  - For RSAs, at the point of acceptance (& payment) by the recipient, the shares should _also_ appear in the relevant Stock Class ledger. (i.e. In OCX exports, visually treat them very similar to an early-exercised option grant).
  - For RSUs, it's the point of release by the company that shares should appear in the relevant Stock Class ledger.
- _Ideally_, equity plan securities should be considered an active element in the cap table one of `Board Approval` or `Issuance` takes place. Board approval _should_ precede issuance, but the emphasis is on capturing truth over proper/expected patterns.
  - The overarching goal is to prevent "double dipping" in the pool - anything that is already moving towards becoming an active security should be considered spoken for.
- Since OCF still requires an Issuance transaction to define a new security in the cap table, there won't be an issue with generating an OCX export based on the above, but for future versions of OCF it's probably worth considering how we could more flexibly capture what action lead to a security's addition to the cap table. 
![IMG_8025](https://github.com/user-attachments/assets/94f8c27b-80c0-49fb-9e53-2e4b9c506f32)

**Comment by jarrettq** [comment]
One question for the TWG folks - when RSAs stipulate that payment is required, do we envision the `Acceptance` event to reflect the recipient counter-signing **AND** paying, or would `Acceptance` reflect signing while `Release` would reflect that the company received payment and is considering the shares Outstanding on the cap table?

**Comment by JSv4** [comment]
While it's not explicit (and should be) - release to me was designed specifically for RSUs. The release is additional step required to actually flip the RSU over to outstanding shares. An RSA, to me, is outstanding the moment it's accepted. 

So, for RSAs:

Issuance --> Acceptance --x Outstanding

For RSUs:

Issuance --> Acceptance --> Release --x Outstanding

**Comment by JSv4** [comment]
It sounds to me that we should also create a document showing very clearly the lifecycle events for different issuance types? @arthur-clara has been doing great work on capturing this implicitly via the xstate validator. Having a reference in the documentation is a must.

**Comment by JSv4** [comment]
We were discussing this again - Acceptance event needs clarity on whether it's an actual legal concept. Appears not to be except for RSAs.
